### ISSUE TITLE
accuracy not changing when using decisionet of NiN

### ISSUE THREAD

**Issue by ItamarDabush** [noamgot/DecisioNet#1]
hi noam,
אני מריץ את הקוד עם הbaseline שבאמת נותן דיוק של 87 אחוז בסוף ההרצה לאחר מכן בהרצה של decision-net נראה שלא מתרחש שינוי בדיוק אני מתחיל לבדוק את זה אבל משער לעצמי שזה אולי קשום לאחד מהhyper parameters או משהו כזה 

זה נשמע הגיוני?

תודה!
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.844%% (2079/21120); Sigma Loss: 0.409 | Sigma Acc: 25.014%% (5283/21120)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.850%% (2093/21248); Sigma Loss: 0.410 | Sigma Acc: 24.995%% (5311/21248)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.857%% (2107/21376); Sigma Loss: 0.409 | Sigma Acc: 25.005%% (5345/21376)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.849%% (2118/21504); Sigma Loss: 0.410 | Sigma Acc: 25.009%% (5378/21504)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.870%% (2135/21632); Sigma Loss: 0.411 | Sigma Acc: 24.991%% (5406/21632)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.876%% (2149/21760); Sigma Loss: 0.412 | Sigma Acc: 24.977%% (5435/21760)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.864%% (2159/21888); Sigma Loss: 0.411 | Sigma Acc: 24.941%% (5459/21888)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.861%% (2171/22016); Sigma Loss: 0.411 | Sigma Acc: 24.918%% (5486/22016)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.849%% (2181/22144); Sigma Loss: 0.411 | Sigma Acc: 24.950%% (5525/22144)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.820%% (2187/22272); Sigma Loss: 0.411 | Sigma Acc: 24.942%% (5555/22272)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.821%% (2200/22400); Sigma Loss: 0.412 | Sigma Acc: 24.938%% (5586/22400)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.814%% (2211/22528); Sigma Loss: 0.413 | Sigma Acc: 24.969%% (5625/22528)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.812%% (2223/22656); Sigma Loss: 0.413 | Sigma Acc: 24.982%% (5660/22656)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.853%% (2245/22784); Sigma Loss: 0.412 | Sigma Acc: 24.978%% (5691/22784)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.873%% (2262/22912); Sigma Loss: 0.413 | Sigma Acc: 24.965%% (5720/22912)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.861%% (2272/23040); Sigma Loss: 0.412 | Sigma Acc: 24.939%% (5746/23040)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.863%% (2285/23168); Sigma Loss: 0.412 | Sigma Acc: 24.953%% (5781/23168)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.864%% (2298/23296); Sigma Loss: 0.411 | Sigma Acc: 24.944%% (5811/23296)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.866%% (2311/23424); Sigma Loss: 0.411 | Sigma Acc: 24.927%% (5839/23424)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.880%% (2327/23552); Sigma Loss: 0.411 | Sigma Acc: 24.932%% (5872/23552)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.886%% (2341/23680); Sigma Loss: 0.412 | Sigma Acc: 24.924%% (5902/23680)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.892%% (2355/23808); Sigma Loss: 0.411 | Sigma Acc: 24.912%% (5931/23808)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.889%% (2367/23936); Sigma Loss: 0.412 | Sigma Acc: 24.912%% (5963/23936)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.890%% (2380/24064); Sigma Loss: 0.411 | Sigma Acc: 24.925%% (5998/24064)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.879%% (2390/24192); Sigma Loss: 0.412 | Sigma Acc: 24.901%% (6024/24192)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.860%% (2398/24320); Sigma Loss: 0.411 | Sigma Acc: 24.905%% (6057/24320)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.886%% (2417/24448); Sigma Loss: 0.410 | Sigma Acc: 24.906%% (6089/24448)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.896%% (2432/24576); Sigma Loss: 0.410 | Sigma Acc: 24.870%% (6112/24576)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.869%% (2438/24704); Sigma Loss: 0.409 | Sigma Acc: 24.870%% (6144/24704)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.862%% (2449/24832); Sigma Loss: 0.410 | Sigma Acc: 24.871%% (6176/24832)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.872%% (2464/24960); Sigma Loss: 0.409 | Sigma Acc: 24.888%% (6212/24960)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.873%% (2477/25088); Sigma Loss: 0.410 | Sigma Acc: 24.880%% (6242/25088)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.883%% (2492/25216); Sigma Loss: 0.410 | Sigma Acc: 24.853%% (6267/25216)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.880%% (2504/25344); Sigma Loss: 0.410 | Sigma Acc: 24.850%% (6298/25344)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.878%% (2516/25472); Sigma Loss: 0.410 | Sigma Acc: 24.839%% (6327/25472)
Total Loss: 2.303; Cls Loss: 2.303 | Cls Top-1 Acc: 9.867%% (2526/25600); Sigma Loss: 0.410 | Sigma Acc: 24.867%% (6366/25600)

**Comment by noamgot** [comment]
Hi @ItamarDabush 
Thanks for using my code (it actually excites me 😅 )
I'll look into it later this week if time will allow.
Obviously, I expect the same behavior as what was presented in the paper. it's been more than a year since I touched this code, so it will take me some time to get what's going on there :)

**Comment by noamgot** [comment]
ok, I think I got this - try running with `--beta 0.5` (that's the value used in our final results in the paper). The default value of beta is 0, so it makes sense that the loss is stuck with this value.
I just ran it and after 3 epochs I'm > 28% top-1 accuracy (in both the train & test sets). If the other hyper-parameters' default values are as we described in the paper, you should end up at ~86.8% test-set top-1 accuracy, as reported in the paper.

**Comment by ItamarDabush** [comment]
thanks! that was the problem. its intresting thats beta =0 makes the network not to improve at all🤔

**Comment by noamgot** [comment]
If I remember correctly, when beta=0 the net collapses to a single computational route, which is just too small for this task. We saw some interesting phenomenons that invite some more research, go ahead :)
